_stbt/core.py

```diff
@@ -1524,6 +1524,11 @@ def push_sample(self, sample):
                 color = (int(255 * max([1 - age, 0.5])),) * 3
                 _draw_text(img, x['text'], origin, color)
             for match_result in matches:
+                label = os.path.basename(match_result.image)
+                # Slightly above the match annotation
+                label_loc = (match_result.region.x, match_result.region.y - 10)
```

Comment:
I considered writing logic to check the frame boundaries and to draw the text on the bottom of the annotation if too near the top or aligned to the right of the annotation if too far to the right. I'd basically hacked this into my installed version because I needed help with debugging and didn't need edge-case handling (pun unintended), but I thought it might be useful to quickly put the patch together to push up (thus no tests either) to get feedback.